toolchain/lowering/lowering_handle.cpp

```diff
@@ -259,7 +273,17 @@ auto LoweringHandleStructValue(LoweringFunctionContext& context,
 auto LoweringHandleStubReference(LoweringFunctionContext& context,
                                  SemanticsNodeId node_id, SemanticsNode node)
     -> void {
-  context.SetLocal(node_id, context.GetLocal(node.GetAsStubReference()));
+  if (node.type_id() == SemanticsTypeId::TypeType) {
+    if (!context.TryToReuseLocal(node_id)) {
+      llvm::Value* v = llvm::ConstantInt::get(context.builder().getInt1Ty(),
+                                              node.GetAsStubReference().index);
+      context.SetLocal(node_id, v);
+    } else {
+      context.SetLocal(node_id, context.GetLocal(node.GetAsStubReference()));
+    }
```

Comment:
What's the purpose of the TryToReuseLocal call here, and which test covers the behavior?

Reply:
There were two different cases that made me add this function: 
When executing line#282 for typetype, it was getting carbon_check with the message "Index doesn't exist".  To remove this, I modified line#282 in the following manner `context.SetLocal(node_id, v);`. But then it gets the carbon check "Duplicate entry ".  And so I added this TryToReuseLocal function. 
The test named two_entries.carbon under the lowering/testdata/tuple folder covers this.

Reply:
I've tried commenting out this code:

```
    // if (!context.TryToReuseLocal(node_id)) {
      llvm::Value* v = llvm::ConstantInt::get(context.builder().getInt1Ty(),
                                              node.GetAsStubReference().index);
      context.SetLocal(node_id, v);
    // } else {
    //   context.SetLocal(node_id, context.GetLocal(node.GetAsStubReference()));
    // }
```

All the tests still pass. I don't think this code is needed, because node_id is the stub reference here and the stub reference should only be in the IR once.

Can you please either remove this code, or add a test which fails without it?